evals/elsuite/audio/eval.py

```diff
@@ -0,0 +1,301 @@
+import base64
+import logging
+import string
+from collections import Counter
+from io import BytesIO
+from typing import Any, List, Optional
+from urllib.parse import parse_qs, urlparse
+
+import jiwer
+import soundfile as sf
+from datasets import Audio, load_dataset
+from pydantic import BaseModel
+from sacrebleu.metrics.bleu import BLEU
+
+import evals
+import evals.metrics
+from evals.api import CompletionFn
+from evals.elsuite.modelgraded.classify_utils import classify
+from evals.record import RecorderBase
+
+logger = logging.getLogger(__name__)
+DEFAULT_SAMPLE_RATE = 16000
+
+
+class Sample(BaseModel):
+    audio: Any
+    transcript: str
+    expected: str
+    context: Optional[str] = None
+
+
+class AudioTask(evals.Eval):
+    def __init__(
+        self,
+        completion_fns: list[CompletionFn],
+        dataset: str,
+        text_only: bool = False,
+        temperature: float = 0.7,
+        max_tokens: int = 4096,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(completion_fns, *args, **kwargs)
+        assert len(completion_fns) == 1, "Audio tasks only support one completion fn"
+        self.dataset = dataset
+        self.text_only = text_only
+        self.temperature = temperature
+        self.max_tokens = max_tokens
+
+    def eval_sample(self, sample: Sample, rng):
+        assert isinstance(sample, Sample)
+        task_prompt = self.build_prompt(sample)
```

Comment:
What if `build_prompt` uses `text_only` as it sees fit?

Reply:
In general I feel like it might be better to let the whole `messages` or `prompt` list be generated by the dataset so it's all in one place. This parent function should likely just take care of the base encoding of the audio and getting it ready for the API.